rholang/src/rust/interpreter/compiler/normalizer/parser.rs

```diff
@@ -8,3 +18,1240 @@ pub fn parse_rholang_code(code: &str) -> Tree {
+                                None => Err(InterpreterError::ParserError(
+                                    "Expected a _proc node in non_empty_cont at index 1"
+                                        .to_string(),
+                                )),
+                            }?;
+
+                            SyncSendCont::NonEmpty {
+                                proc: Box::new(parse_proc(&proc_node, source)?),
+                                line_num: choice_node.start_position().row,
+                                col_num: choice_node.start_position().column,
+                            }
+                        }
+
+                        _ => {
+                            return Err(InterpreterError::ParserError(format!(
+                                "Unexpected choice node kind: {:?} of sync_send_cont",
+                                choice_node.kind(),
+                            )))
+                        }
+                    },
+                    None => {
+                        return Err(InterpreterError::ParserError(format!(
+                            "No child choice node at index 0 for sync_send_cont",
+                        )))
+                    }
+                }
+            };
+
+            Ok(Proc::SendSync {
+                name: name_proc,
+                messages: messages_proc,
+                cont,
+                line_num,
+                col_num,
+            })
+        }
+
+        "new" => {
+            let decls_node = get_child_by_field_name(node, "decls")?;
+            let mut name_decls = Vec::new();
+
+            for child in decls_node.named_children(&mut decls_node.walk()) {
+                let var_node = child
+                    .child(0)
+                    .filter(|n| n.kind() == "var")
+                    .ok_or_else(|| {
+                        InterpreterError::ParserError(
+                            "Expected a var node in name_decl at index 0".to_string(),
+                        )
+                    })?;
+                let var_proc = parse_var(&var_node, source)?;
+
+                let uri = match child.child_by_field_name("uri") {
+                    Some(uri_literal_node) => {
+                        match uri_literal_node.next_sibling() {
+                            // 'uri_literal_node' is '(' so it's sibling is the actual rule we want
+                            Some(rule_node) => Some(parse_uri_literal(&rule_node, source)?),
+                            None => {
+                                return Err(InterpreterError::ParserError(
+                                    "Expected rule on uri_literal node".to_string(),
+                                ))
+                            }
+                        }
+                    }
+                    None => None,
+                };
+
+                name_decls.push(NameDecl {
+                    var: var_proc,
+                    uri,
```

Comment:
I think it may be better to keep it separate because in the grammar, `_proc` in turn can just be a `uri_literal`. So there may be a case where you have a `uri_literal` without a `var`. And then of course cases where you just have a `var` without `uri`. I could add an `uri` option field to `var` but I think this would only be `Some` under `name_decl`.

Reply:
Well, yes, but these use cases are different - one is var definition (var is bound to a `registry`), the other is simply a value. I do not have a strong opinion here, but the former option seems to be very tightly bonded - a var with uri is something entriely different than "just" a var